### ISSUE TITLE
I think the fire code won't work right when CNDV is used along with transient crop areas

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/ctsm#52]
**Bill Sacks < sacks@ucar.edu > - 2016-12-21 09:21:21 -0700**
**Bugzilla Id:** [2392](http://bugz.cgd.ucar.edu/show_bug.cgi?id=2392)
**Bugzilla CC:**  dlawren@ucar.edu,  rfisher@ucar.edu, 

Fang Li's fire code (CNFireXXX) has code to handle deforestation-related fires. I think these are meant to be associated with human-induced land cover change. When the fire code was first written, I believe CLM only allowed one of transient PFTs or CNDV to be active at a given time. However, moving forward we want to allow CNDV at the same time as prescribed transient crop areas. We currently pass the full area changes from all sources to the fire code. This means that (e.g.) if there is a decrease in tropical forest area due to prescribed land cover change, and a further decrease from CNDV, the fire code will see the total decrease and base deforestation-related fires on this total decrease. I *think* what we really want is for the fire code to just see the decrease from prescribed land cover change, since this represents human-induced deforestation.

There is a similar problem in theory from changes in glacier area: decreases in forest area due to expanding glaciers are seen the same way as human-induced land cover change by the fire code. However, in practice, we don't expect glacier expansion to occur in regions of tropical forests. But this same issue could also arise from future, internally-generated land cover change, such as the expansion of lakes / wetlands in inundated areas.

Off-hand, I can see two solutions to this problem:

(1) Separately track land cover change (dwt) from human sources. This would include changes in areas generated in dynpft_interp and dyncrop_interp, but not from other sources. This is a bit tricky because of these multiple streams of human-induced land cover change, and the fact that it can manifest as changes in pft%wtcol and/or col%wtgcell. I imagine this would be doable, but at the cost of some additional code complexity.

(2) Rather than trying to deduce the human induced land cover change from the various sources of land cover change in the model, instead have an explicit field specifying the rate of human-induced deforestation on the surface dataset and landuse_timeseries file - or specified as a stream. This would be a straightforward approach, though we'd probably want to construct this field to be consistent with the actual land cover change in the model - and for that reason, may want it on the landuse_timeseries file rather than as a separate stream. One advantage of this approach is that it would provide a mechanism to specify rates of land cover change in a year-2000 run (i.e., a present day run, which has no land cover change). Another advantage is that gross transitions (rather than just net transitions) could be included in this field. (Note that if we wanted to go down this path, then it could make sense to specify a deforestation rate more generally in the code for a year-2000 run - i.e., used in other parts of the code than just the fire code.)

**Comment by ekluzek** [comment]
Since, we've deprecated CNDV, I'm closing this as a wontfix.

**Comment by billsacks** [comment]
We closed this as as wontfix because CNDV is deprecated, but I realized that similar issues exist with transient lakes and glaciers combined with transient crops. Initially the issue had been named too specifically; I have renamed it to include those other problems.

That said, I see this issue as minor and I don't see this as something that will realistically get fixed anytime soon, so I'm going to keep it closed as a wontfix.
